### DISCUSSION TITLE
Swedish Language Leader

### DISCUSSION THREAD

**Discussion by synesthesiam** [OHF-Voice/intents#406] [Language Leaders]
@lellky are you interested in being the Swedish language leader? Another person from Discord has volunteered; I'll point them to this issue.

**Comment by edgein666** [comment]
Hi, the person from Discord here.. 🙂 let me know if I can help. BR Björn

**Comment by larsdunemark** [comment]
I can also help out with this if there is need for multiple workers or a work group.

**Comment by lellky** [comment]
Good morning!

I am interested, and it would be an honor, but I have two younglings here at home and I don't know if I have the time to fully commit to be the leader. I will gladly help with reviews and have discussions though.

Also, there is another user _thedailyuncle_ on discord: https://discordapp.com/channels/330944238910963714/1054846248374505572/1056591259944370186

**Comment by synesthesiam** [comment]
No worries, @lellky. Is anyone else interested in being a Swedish language leader? I've invited "thedailyuncle" from Discord to join us here too.

**Reply by larsdunemark** [reply]
I can take the role of language leader for Swedish, if someone else would also like to be involved, I think it would be great to share the work with me because it would be my first assignment in an open source project.

**Reply by edgein666** [reply]
I have nothing against you taking the role Lars and I would be happy to share the work with you. We could get the work going with some Discord calls maybe? I don't have much experience with GitHub and open source projects but we could probably work it out together. BR Björn

**Comment by ph13t0n** [comment]
Hi Synesthesiam,

I hope this message finds you well. I am writing because I previously expressed my interest in the open position as a translator. I am confident that my skills and experience make me a strong fit for the role.

Thank you for considering my application. I look forward to the opportunity to contribute to your team.

Sincerely,
Per Rudhamn, thedailyuncle#5220 @ discord.

**Reply by edgein666** [reply]
Hi Per, 

Just to double check, have you seen the release party yesterday where they asked the community for language leaders? 

If not; the leader would get the responsibility to validate and approve the submissions from, I guess, anyone who wants to contribute.

Someone please correct me if I misunderstood.

So, if you just want to translate, please go ahead. If you want to be the leader, maybe check with Lars who has also shown interest.

My suggestion is that we try and work as a team but with one leader. As I mentioned in a previous comment I am myself not sure how all of this works but I'm happy to help out.

BR Björn

**Reply by ph13t0n** [reply]
Hi Björn! 

You seems suitable for the task, i only responded to a reply and announced that I'm here. 

All the tasks, events and newsletters who has been released from HQ, has been reviewed and fully understand in what will come, in the near time and in the future and I guess, with all good intentions that a leader should bring people, is to get them together, lift and inspire members to do good and as you, suggested - maybe a little, harsh - I can not lead when the first suggestion and response is to go away and "just translate" when this is my first contact to a great team. 

And as you probably know, every bit counts, to give the end-user a fantastic experience and enjoy a great version of Home Assistant, now and forever. 

That is my beacon, that is my goal with everything I do. 

I go forward and "just translate" and skip the leadership -  may the force be strong! 

//Per Rudhamn

**Comment by synesthesiam** [comment]
The language leader(s) will be responsible for reviewing and merging PRs. Of course, they can contribute translations as well :slightly_smiling_face:

**Reply by larsdunemark** [reply]
I feel like handling PR and reviewing shouldn't be an issue, finding the right level for the linguistic part may be my bigger challenge and I think that it will benefit from multiple reviewers to get it as smooth as possible.

**Reply by ph13t0n** [reply]
Lars! 

well written - you can't spell team with an "I", it's impossible. Enjoy the ride //Per

**Reply by Robin-St** [reply]
I think the part of handling the level of the linguistics is a good part for group collaboration. Whenever in doubt just start a discussion at the relevant PR and tag those of us that usually add insights.

**Comment by synesthesiam** [comment]
Just merged a PR from @Robin-St. Tagging them here for language leader discussion.

**Reply by Robin-St** [reply]
Thanks, I thought about applying but think I have to pass on formal responsibilities for now. But I’m really interested in helping with discussions and writing intents. (And down the road start to test and use HA with local voice :) )

**Reply by edgein666** [reply]
Great work @Robin-St! I saw your translation and also that @larsdunemark has pushed something.

Would anyone of you mind showing me your workflow? I have vscode and I am using GitHub with my personal account. The files with the translations seems pretty straight forward. But I have some gaps and would appreciate some help connecting the dots.

Discord EdGe#8257

**Reply by Robin-St** [reply]
@edgein666 I just followed this https://github.com/home-assistant/intents/blob/main/docs/codespace/README.md to set up a codespace. It was really easy and convenient and gave a virtual machine running at github with all dependencies and vscode right in the browser, but from the top left menu you can choose to open it in the desktop editor instead which worked great (still connected to the VM in the cloud).

**Reply by edgein666** [reply]
I saw your post but never read the article, was so sure that there would be a cost for me personaly. Just checked and this seems like the way to go for me. Thanks :)

**Comment by edgein666** [comment]
Just made PR #499 but not sure it's done correctly. Comments appreciated. :)

**Reply by Robin-St** [reply]
The added intent variations looks good but the bot seems to be upset with some git technicalities. I'm guessing that the commits are done by a local user and not your gitlab user. Did you use a codespace or check out the code locally?

**Reply by edgein666** [reply]
Yeah, I had one issue with the mail-address, so added the additional address, then I had to sign some "CLA". Both should be fixed. Might just be that I need to be approved by a person? "1 workflow awaiting approval"

I made a fork, opened it with vscode, then pushed the changes and then clicked on "create pull request" in the github webpage, that created #499 and then I had to add the mail and the CLA as mentioned above.

Maybe I'm good and just need to wait for @synesthesiam or someone to approve me?

**Reply by Robin-St** [reply]
Partly good. The addition to _common.yaml works right away. But for the light set you have added good tests and found a "ändra" but you have missed to update the actual intent. The intent is coded in https://github.com/home-assistant/intents/blob/main/sentences/sv/light_HassLightSet.yaml and to that file "ändra" needs to be added to the list of words to use. 

There are test scripts that you can run that will test each sentence added to the tests in the test folder against the intents definitions in the sentences folder. Whit these it's easy to add more tests sentences and continuously add or tweak intent definitions until all test sentences are correctly matched :)

**Reply by edgein666** [reply]
Right, didn't think about that. I will look into it, thanks!

**Comment by edgein666** [comment]
Question about:
 - "[sätt | vrid | dra [på] | skruva | dimmra] [upp | ner ] <name>[s | n | ns ] [<ljusintensitet>] till <brightness>"

in:
https://github.com/home-assistant/intents/blob/main/sentences/sv/light_HassLightSet.yaml

Are, let's say, [upp | ner ] mandatory? That is; either "upp" or "ner"? Can it be silent? If not, how can we accomplish that?

//Björn

**Reply by Robin-St** [reply]
"[ ]" is optional and "( )" mandatory.
https://github.com/home-assistant/intents/blob/main/sentences/README.md#sentence-templates

**Reply by edgein666** [reply]
Btw, shouldn't we use variables in this file as well? Like we have in the on/off files? In the name of consistency.

**Reply by Robin-St** [reply]
Yes, i started to do that in a pull request that haven been merged yet. https://github.com/home-assistant/intents/pull/487

**Comment by Robin-St** [comment]
Regarding entity id for test fixtures, there where a mix of English and Swedish names for the entity id's for the fixtures both areas and items. I made a pull request that has the names in Swedish but all id's in English, as it feels more natural to have "variable" names in English for me. The same pull request also clears up some duplicate areas for different endings for kitchen (as the different spellings had different id the tests started to fail when i added logic to handle all different endings for rooms)

Does anyone here have an opinion regarding the language of the id's? I can remake them in Swedish if we prefer that, i just want it consistent :) 
https://github.com/home-assistant/intents/pull/497

**Reply by edgein666** [reply]
I agree, English and consistent. 👍

**Reply by lellky** [reply]
I also agree. The id's could be in english (although my id's in my home assistant instance are in Swedish).

**Comment by lellky** [comment]
A general question about naming devices/entities/areas;

In my Home Assistant I have areas (in Swedish) like:
* Vardagsrummet (the living room)
* Köket (the kitchen)
* Sovrummet (the bedroom)

and entities/devices like:
* Kökslampan (the kitchen lamp)
* Taklampan i vardagsrummet (the ceiling lamp in the living room)
* Golvlampan i vardagsrummet (the floor lamp in the living room)
* Temperatur i sovrummet (the temperature in the bedroom)
* Jonas sänglampa (Jonas's bedside lamp)

Do we know how people in general name their devices/entities/areas and how will this be handled in general considering the examples above containing the area (in the name)? Also some people seem to name their areas in definite form and others in indefinite form (vardagsrummet/the living room v.s. vardagsrum/living room).

**Reply by Robin-St** [reply]
I would guess that there are a mix of detailed naming as your and more loose naming. With the recent ability to have multiple names for entities i would think that there might be good cases to name a lamp "Taklampa Sovrum" and "Taklampa" just to be able to address it by a unique name or a combination of area and not unique name.

Regarding naming areas, i hope that the following rules will expand most (all?) room names but have not done extensive testing:
```
  name: "{name}[s | n | ns | en ]"
  area: "{area}[e | me][t | ts | n | ns ]"
```

**Comment by edgein666** [comment]
Hi,

trying to understand git and github. right now we have 5 Open pull requests of different files, can I open the same file and modify it or will something get messed up with the versioning? I guess git is done for this but I'm clueless about how it should work.

BR Björn

**Reply by larsdunemark** [reply]
When a commit is merged the changes in that commit will be added to the main branch. Before that it doesn't matter. But if there is another commit modifying the same file there will be a merge conflict that will need manually correction. This is usually fixed by rebase your commit on the latest main branch.

**Reply by edgein666** [reply]
Ok, thanks for the explanation. 

So if I want to change one of the files currently in the queue to be approved and merged, should I wait until they are merged and do a new pull (?) and then make my change?

What is the best approach?

**Reply by Robin-St** [reply]
Best approach is a tricky question. The risk of waiting for the right opportunity is to always miss "the window" and the risk of not waiting is the potential added work to merge. It is easier to add things to the same file than it is to modify the same lines, just something to get a feel for :) But to get going i would say not to wait, build more momentum and get up to speed and then we can help resolve the merges. 

If you want to see live example look at https://github.com/home-assistant/intents/pull/487. The first two commits where before i opened the PR and all tests against main where running ok. Then https://github.com/home-assistant/intents/pull/497 that where submitted after got merged into main, making the modification not work and the tests fail. Then opened up that branch in vs-code again and pulled latest main and selected to merge that into this branch. While doing so git merges the changes that have no conflicts and asks for the changes that where done to the same places of the code which to keep.

With the changes from main now in the branch i ran the tests, looked at the errors and fixed the compatibility so that the change now works with the new main. Buy just committing these changes to the branch they are visible from the PR and the tests now run again :D

**Reply by edgein666** [reply]
Still figuring things out.. Did #580 now. Let me know what/if something is wrong. :)

**Reply by lellky** [reply]
I recommend reading this article: https://medium.com/singlestone/a-git-workflow-using-rebase-1b1210de83e5

Also, keep in mind that the more open pull requests the more complexity and risk for merge conflicts. It is always a good idea to make small changes in a pull request and try to get them merged before opening new pull requests. Also rebasing on upstream (main repo) is a must and is described in the article linked above.

**Comment by synesthesiam** [comment]
@larsdunemark adding you as language leader :+1:

**Reply by larsdunemark** [reply]
Thanks for the opportunity, does this mean that I should start merging all MR that looks good for Swedish, and is there anyone that I can notify for feedback on my on commits since it doesn't feel right to approve and merger my on stuff without additional reviews. :)

**Reply by Robin-St** [reply]
@larsdunemark If you want some second eyes for the language part of your commits you can ping me, if it helps :)

**Reply by lellky** [reply]
You can ping me as well.

**Reply by edgein666** [reply]
I'm also trying to be here. :)

**Reply by synesthesiam** [reply]
@larsdunemark You're free to merge PRs that look good, but I would suggesting tagging others for feedback :slightly_smiling_face:

**Reply by larsdunemark** [reply]
It doesn't look like I can take a person that is not part of the HA group or not already part for the MR so if anyone would like to put a comment on my MR https://github.com/home-assistant/intents/pull/433 it would be great.

**Comment by Robin-St** [comment]
Just for fun i tested the sampling feature to see how many sentences are recognized. For HassTurnOn we have 56035 different sentences with all variations of letter, a lot of these sentences does not make sense, but still a huge number. Also made the mistake to run it for all intentions, but as it generated a sentence for each variation of the commands for each set brightness percentage i had to stop the script

**Reply by edgein666** [reply]
Swenglish: Many variations it will become. :)

**Reply by synesthesiam** [reply]
Keep in mind that the goal for this repo is to just have very common ways of saying things. Users will always have the option to create custom sentences.

**Reply by edgein666** [reply]
thanks @synesthesiam very good to know!

**Comment by djungelola** [comment]
Hi, my first test of Assist was turning on a light and my first choice was "Tänd lampan i köket" but didnt work.
Saw in the yaml that "Slå på/av" is what was defined. Possible to add "Tänd" and "Släck" for lights?
That feels more natrual to me atleast.

/Ola

**Reply by larsdunemark** [reply]
Hi @djungelola 
We are using the template variable {slå_på} and {slå_av} that should resolve to contain "Tänd" and "Släck"
https://github.com/home-assistant/intents/blob/main/sentences/sv/_common.yaml#L61-L62
I started testing it out for some parts last night but haven't had time to verify all scenarios. It seems that there might be some issues left when matching sentences.
Did it work with "slå på lampan i köket" or did that also fail?

**Reply by lellky** [reply]
Is your Home Assistant running in Swedish or English? It seems to work only in the language set for Home Assistant, i.e. if Home Assistant is set to English the assist feature will only recognise English input.

**Comment by djungelola** [comment]
Yes, you are right!
"Tänd/Släck alla lampor i köket" works.
"Slå på/av alla lampor" i köket works.
"Tänd/Släck lamporna i köket" doesnt work, but might be as designed maybe?

Either way, thanks for driving this in Swedish!!

Br Ola

**Reply by edgein666** [reply]
that is strange..

"släck/tänd lamporna i köket" works for me!

**Comment by djungelola** [comment]
it worked for me now as well, maybe SBS on my side :-)
Will continue testing.

**Reply by lellky** [reply]
Glad to hear!

**Comment by mrdanielk1** [comment]
Hello, fellow Swedes,

I'm having big problems getting Assist to understand what I type (and also say). As soon as I add an area to the sentence, it fails. Either with "Ett oväntat fel.." or "ursäkta, jag förstår inte".

My setup is fairly basic, I have some rooms, and in the rooms I have some devices. Nothing fancy. Most of the devices are connected through Telldus. Not that it should matter. Language is set to Swedish.

Anyway, I have window lamps in several rooms. They are all named "Fönsterlampa". But they are connected to different areas. The areas are named "Kök", Allrum" and so on.

In Assist, If I type "Tänd fönsterlampan i köket", I would expect the window light in the kitchen to be activated, but it fails with "Ursäkta, jag förstår inte". If I omit the area, and type: "Tänd fönsterlampan", it does turn one of the window lights on, but I can't control which one.

I have tried all kinds of tricks to get it to work, but the areas always make it fail. I've tried all kinds of grammar differences to try to fix it, like "Tänd fönsterlampa i kök", but it makes no difference. I've also tried simple phrases like "Tänd alla lampor i kök", but those cause "Oväntat fel..."

Any ideas what I'm doing wrong?

Thanks in advance.
/Daniel.

**Reply by edgein666** [reply]
Hi,

first of all, thanks for testing!

I tried "tänd sänglampan på ovanvåningen" which is similar to your sentence and it doesn't work either.

But "tänd sänglampan" works.

I guess the intent should match a sentence in this file:
https://github.com/home-assistant/intents/blob/main/sentences/sv/light_HassTurnOn.yaml

But I cannot find it.

I think we need to add something like:

`"<slå_på> <name> <i_på> <area>"`

these are the variables:

```
slå_på: "(tänd | sätt (på | igång | fart på) | slå (på | till) | starta)"
name: "{name}[s|n|ns|en]"
i_på: "(i | på | vid | bredvid)"
area: "{area}[e|me][t|n][s]"
```

also pinging @larsdunemark  and @Robin-St  for comments.

//B

**Reply by edgein666** [reply]
I have added "[SV] Possibility to turn on light in area. #933" to solve this.

**Reply by mrdanielk1** [reply]
Awesome. Thanks for the quick response. :)

I have looked at the files and assumed that ` "<slå_på> [<alla>] <ljuskällor> <i_på> <area>"` would cover my use case, since I assumed `[<alla>]` was optional. But I was clearly wrong. 

However, "Tänd alla lampor i köket" also fails for me with "Oväntat fel...". @djungelola (a few posts up) seems to have success with that. I cannot refer to any area with "alla lampor"

So I'm still feeling that my areas cause problems...

Thanks again!

/D

**Reply by mrdanielk1** [reply]
Quick update, some areas seem to work. "allrummet" and "sovrummet" works with "alla lampor". "vardagsrummet", "köket" and "arbetsrummet" does not...

Strange...

**Comment by synesthesiam** [comment]
@larsdunemark when you get a chance, could you please look into translating the `HassGetState` sentences/responses?

**Comment by slungs** [comment]
Hi, is @larsdunemark still acting as language leader for Swedish? I just got my voice PE and the experience with swedish... well, still needs some improvement. Happy to help in any way possible.

/Anton

**Reply by larsdunemark** [reply]
Sure is, if you find something that you are missing and have the possibility to add a PR I will make sure that we can merge it.

**Comment by gurrasv** [comment]
I would love to help out anyway i can.

I have an accent from the southern part of Sweden, might be somewhat of help.

**Reply by larsdunemark** [reply]
If you have any missing sentences please open an pull request and we can make sure to get it in.

**Reply by gurrasv** [reply]
I would go more towards thats its an accent issue rather than sentence issue though

**Reply by larsdunemark** [reply]
Ok, this project is just the intent sentence, are you problem related to speech to text or that sentence are missing?

**Comment by gurrasv** [comment]
Im having issues with the speech to text part mainly. It simply does not
understand me 50% of the time and almost all of the time if im not using a
LLM as conversation agent.

With a LLM av conversation agent, it seems to not be needing to be perfect,
the LLM understands the context much better.

But Yeah STT issues are definitely the biggest part in why im not moving
from Google Assistant to using assist in homeassistant as the primary voice
control.

Would love to help out anyway i can!!

Best regards, Gustav
(om du föredrar att vi pratar svenska så kan vi göra det)

Den tis 25 feb. 2025 23:04Lars Dunemark ***@***.***> skrev:

**Comment by gurrasv** [comment]
You still there?

Den tis 25 feb. 2025 23:04Lars Dunemark ***@***.***> skrev:

**Reply by larsdunemark** [reply]
Sure, got hold up on my replay, since this project is only for sentences and not STT we need to get more information on what the STT is generating that does not match the current intents.

In case you are using a STT you can go in to the assistant in HA and show debug data with what it sent for intent and compare what you expected.

Since we still waits for speech to phrase for Swedish we don't know if it will make some simples intents to work better. 
A STT that would be aware on available sentences would probably be better to match valid commands but would probably also end up that invalid sentences would result in a valid command for something that the user did not intend to trigger.

In case of missing sentences you can ether open an issue, or a PR and we can work from that. Please make sure that changes are keep small and in one area at the time to make it easier to review and integrate if there would be future conflicts.

**Comment by gurrasv** [comment]
I just noticed that the timer function for Swedish is no longer working.

I tried everything, i debugged everything i said and even used the text
thing to try my commands out before trying to actually say it.

I tried:
"Sätt en timer på 10 sekunder"
"Timer 10 sekunder"
"Sätt en 10 sekunder timer"
(And more)

I'm using a LLM as conversation agent with the "prefer handling commands
locally" ticked, did not work. And even with the built in homeassistant
conversation agent it doesnt work.

When changed to English language for all pipeline parts it worked as
expected. This is something that has come and gone before but last 1.5
months it has not worked.



Den tors 6 mars 2025 23:07Lars Dunemark ***@***.***> skrev:

**Reply by larsdunemark** [reply]
ok, using the Development tools -> Assist page in home assistant you can try out what a sentence would match.

LLM will not use the intents that is created in this project, LLM is an "AI" that tries to guess what it should do, I haven't used it my self that much so I don't know how it get the information about available intents or sentences.

But the following sentences works with the build in intents for Swedish:
* "Sätt en timer på 10 sekunder"

```yaml
intent:
  name: HassStartTimer
slots:
  seconds: '10'
details:
  seconds:
    name: seconds
    value: 10
    text: '10'
targets: {}
match: true
sentence_template: <timer_start> [en|ett] <timer_duration>[s] <timer>
unmatched_slots: {}
source: builtin
```

* "Sätt en 10 sekunder timer"

```yaml
intent:
  name: HassStartTimer
slots:
  seconds: '10'
details:
  seconds:
    name: seconds
    value: 10
    text: '10'
targets: {}
match: true
sentence_template: <timer_start> [(en|ett)] <timer> [på ]<timer_duration>
unmatched_slots: {}
source: builtin
```

**Comment by Hedda** [comment]
Anyone working on adding support for Swedish language translation (sv) for all the "[speech-to-phrase](https://github.com/OHF-Voice/speech-to-phrase)" sentences to Home Assistant? See:

* https://github.com/OHF-Voice/speech-to-phrase

and

  * https://github.com/OHF-Voice/speech-to-phrase/issues/100

PS: For reference, Voice chapter 9 brought optoinal "Speech-to-Phrase" support to Home Assistant voice control (not the same as intents):

* https://www.home-assistant.io/blog/2025/02/13/voice-chapter-9-speech-to-phrase/

  * https://www.youtube.com/watch?v=k6VvzDSI8RU

**Reply by Hedda** [reply]
FYI, I submitted an initial pull request draft for speech-to-phrase Swedish (Svenska) translation if anyone here would like help review it:

* https://github.com/OHF-Voice/speech-to-phrase/pull/101

**Comment by gurrasv** [comment]
What do you mean?

Den mån 10 nov. 2025 11:26Hedda ***@***.***> skrev:

**Comment by WaterInTheLake** [comment]
@larsdunemark, I know life can get busy, so absolutely no pressure at all. I just wanted to check in to see if you are still active as a language leader? I can see that your GitHub activity is more sparse these days... :-)

**Reply by larsdunemark** [reply]
I announced in the discord channel that I would not be able to continue as language leader from around 6 months ago.  I should probably have followed up on that. I will see if I can review was for review but it would be good to promote a new language leader.

**Comment by Robin-St** [comment]
I noticed that @larsdunemark has requested to step down as language leader for Swedish and I would like to apply for the position. I was interested in applying when the intents launched but did not have the energy back then to take any responsibility so I just contributed sentences. 
I don't know if there are others interested in the position, but I just wanted to ping that I'm available if wanted/needed @synesthesiam

**Reply by synesthesiam** [reply]
Definitely @Robin-St! I'll get you set up.
